pkg/generate/app/cmd/newapp.go

```diff
@@ -506,16 +526,18 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 
 				// Append any exposed ports from Dockerfile to input image
 				if ref.Input().Uses.IsDockerBuild() {
-					exposed, ok := ref.Input().Uses.Info().Dockerfile.GetDirective("EXPOSE")
-					if ok {
-						if input.Info == nil {
-							input.Info = &imageapi.DockerImage{
-								Config: &imageapi.DockerConfig{},
+					if df, err := ref.Input().Uses.Info().Dockerfile.Dockerfile(); err == nil {
+						exposed, ok := df.GetDirective("EXPOSE")
```

Comment:
sorry we do: https://github.com/openshift/origin/blob/1b9f255f61d66d97e644cd283612a816f46aa550/pkg/generate/app/app.go#L369